File: 04_Generating_2bit_Vectors.cpp
#include <iostream>
#include <string>

void generate_bit_vectors(int n, std::string vector) {
    if (n == 0) {
        std::cout << vector << std::endl;
    } else {
        generate_bit_vectors(n - 1, vector + "0");
        generate_bit_vectors(n - 1, vector + "1");
    }
}

int main() {
    int n;
    std::cout << "Въведете брой битове (n): ";
    std::cin >> n;
    generate_bit_vectors(n, "");
    return 0;
}

File: 04_Generating_2bit_Vectors.py
def generate_bit_vectors(n, vector):
    if n == 0:
        print(vector)
    else:
        generate_bit_vectors(n - 1, vector + "0")
        generate_bit_vectors(n - 1, vector + "1")

def main():
    n = int(input())
    generate_bit_vectors(n, "")

main()


